### PR TITLE
Add documentation for async/await.

### PR DESCRIPTION
**Pull request by cristianoc**

### CODE REVIEW

File: pages/docs/manual/latest/function.mdx [hunk 1]
```diff
@@ -415,6 +415,78 @@ If you write down the uncurried function's type, you'll add a dot there as well.
 
 **This feature seems trivial**, but is actually one of our most important features, as a primarily functional language. We encourage you to use it if you'd like to remove any mention of `Curry` runtime in the JS output.
 
+## Async/Await (from version v10.1)
+
+Just as in JS, an async function can be declared by adding `async` before the definition, and `await` can be used in the body of such functions.
+The outpul looks like idiomatic JS:
```
Comment (zth) [inline pages/docs/manual/latest/function.mdx:421]: Typo outpul

File: pages/docs/manual/latest/function.mdx [hunk 2]
```diff
@@ -415,6 +415,78 @@ If you write down the uncurried function's type, you'll add a dot there as well.
...
+```
+```js
+async function greetUser(userId) {
+  var name = await getUserName(userId);
+  return "Hello " + name + "!";
+}
+```
+</CodeTab>
+
+The return type of `getUser` is inferred to be `promise<string>`, where type `promise` is now a built-in of the language.
```
Comment (zth) [inline pages/docs/manual/latest/function.mdx:441]: Not sure we need to mention that the promise type is now built in, might just confuse and it's an implementation detail that most users don't need to know about.

File: pages/docs/manual/latest/function.mdx [hunk 3]
```diff
@@ -415,6 +415,78 @@ If you write down the uncurried function's type, you'll add a dot there as well.
...
+let someAsyncFn = async () => {
+  switch await somethingThatMightThrow() {
+  | data => Some(data)
+  | exception JsError(_) => None
+  | exception SomeReScriptException => None
+  }
+}
+```
+```js
+var SomeReScriptException = /* @__PURE__ */Caml_exceptions.create("JsxV4.SomeReScriptException");
```
Comment (zth) [inline pages/docs/manual/latest/function.mdx:466]: JsxV4 might be confusing here, but not sure we need to care tbh.

**Review by zth — APPROVED** [review]
Great!

File: src/components/Navigation.mjs [hunk 4]
```diff
@@ -602,13 +602,13 @@ function Navigation(Props) {
                             }, React.createElement("div", {
                                   className: "hidden sm:block mr-6"
                                 }, React.createElement(DocSearch.make, {})), React.createElement("a", {
-                                  className: "mr-5 no-underline block text-inherit hover:cursor-pointer hover:text-fire-30 text-gray-40 mb-px",
+                                  className: "mr-5 " + link,
```
Comment (cristianoc) [inline src/components/Navigation.mjs:605]: only inlining difference

**Review by zth — APPROVED** [review]
Awesome! 👍

**Comment by ryyppy** [comment]
I guess there also need to be changes for the Async & Promise section as well?

**Pull request merged**